src/azure-cli-core/azure/cli/core/profiles/_apiversions.py

```diff
@@ -0,0 +1,114 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from functools import total_ordering
+from ._shared import SDKProfile, get_client_class, ResourceType
+
+
+class ApiVersions(object):
+    def __init__(self, api_version_data, resource_type=None):
+        self._resource_type=resource_type
+        self._sdk_profile = None
+        self._op_groups = set()
+
+        if isinstance(api_version_data, str):
+            self._default_version = api_version_data
+
+        elif isinstance(api_version_data, SDKProfile):
+            if not resource_type:
+                raise ValueError('The resource_type is missing.')
+            self._sdk_profile = api_version_data
+            self._default_version = self._sdk_profile.default_api_version
+            self._op_groups = set(
+                [op for op, t in get_client_class(resource_type).__dict__.items() if isinstance(t, property)])
+
+        else:
+            raise ValueError('The given api version data is neither a str or {}'.format(SDKProfile.__name__))
+
+    def __str__(self):
```

Comment:
I'm not a huge fan of hacking `__str__` here. This method is supposed to be an accurate and readable description of the instance, and should not raise an exception if some states are "invalid". I'd prefer an explicit naming "get_single_string", or directly using "get_version" when necessary.